src/hooks/screen/useRootChannelListHook.ts

```diff
@@ -29,6 +40,76 @@ const useRootChannelListHook = () => {
     }
   };
 
+  const requestPermission = async () => {
+    const authStatus = await messaging().requestPermission();
+    const enabled =
+      authStatus === messaging.AuthorizationStatus.AUTHORIZED ||
+      authStatus === messaging.AuthorizationStatus.PROVISIONAL;
+
+    if (enabled) {
+      const fcmToken = await messaging().getToken();
+      const payload = {
+        fcm_token: fcmToken
+      };
+      fcmTokenService(payload);
+    }
+  };
+
+  const checkNotificationPermission = async () => {
+    PushNotification.checkPermissions(
+      (
+        data: PushNotificationPermissions & {
+          authorizationStatus?: number;
+        }
+      ) => {
+        const lastPromptTime = StorageUtils.lastPromptNotification.get();
+        const promptInterval = 48 * 60 * 60 * 1000;
+        const currentTime = new Date().getTime();
+        if (lastPromptTime && currentTime - parseFloat(lastPromptTime) < promptInterval) {
+          return; // Don't show the prompt if the interval has not passed
+        }
+
+        const needsPermission =
+          data.alert === false || data.badge === false || data.sound === false;
+
+        const showAlert = (onPressAction: () => void) =>
+          Alert.alert(
+            "Don't Miss New Messages",
+            `Allow notifications to know 
+when friends send you messages.`,
+            [
+              {text: 'Not now', onPress: () => console.log('Cancel Pressed')},
+              {text: 'Allow', onPress: onPressAction, isPreferred: true}
+            ]
+          );
+
+        switch (data.authorizationStatus) {
+          case PERMISSION_STATUS_ACCEPTED:
+            if (needsPermission) {
+              // THIS CASE IT'S TRIGGERED WHEN USER MANUALLY DISABLED THE NOTIFICATION
+              StorageUtils.lastPromptNotification.set(currentTime.toString());
+              showAlert(() => openSettings().catch(() => console.warn('cannot open settings')));
+            } else {
+              requestPermission();
+            }
+            break;
+          case PERMISSION_STATUS_BLOCKED:
+            if (needsPermission) {
+              StorageUtils.lastPromptNotification.set(currentTime.toString());
+              showAlert(() => openSettings().catch(() => console.warn('cannot open settings')));
+            }
+            break;
+          case PERMISSION_STATUS_PENDING:
+            StorageUtils.lastPromptNotification.set(currentTime.toString());
+            showAlert(() => requestPermission());
+            break;
+          default:
+            break;
+        }
+      }
+    );
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `checkNotificationPermission` function uses console logs to handle errors or user actions. It would be better to replace these with proper error handling or user feedback mechanisms. Also, the function is quite large and does a lot of different things, which goes against the Single Responsibility Principle. Consider breaking it down into smaller, more manageable functions.
```diff
- {text: 'Not now', onPress: () => console.log('Cancel Pressed')},
+ {text: 'Not now', onPress: () => {/* handle user action */}},

- showAlert(() => openSettings().catch(() => console.warn('cannot open settings')));
+ showAlert(() => openSettings().catch((error) => {/* handle error */}));

- showAlert(() => requestPermission());
+ showAlert(() => requestPermission().catch((error) => {/* handle error */}));
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->